astropy/coordinates/representation.py

```diff
@@ -325,14 +325,21 @@ def __rsub__(self, other):
         return self._combine_operation(operator.sub, other, reverse=True)
 
     # Required to check equality between representation and differential
-    # classes
+    # classes.
     def __eq__(self, other):
         if self.__class__ != other.__class__:
             return False
 
+        if self.isscalar and other.isscalar:
```

Comment:
@souravghosh97 - this is very long-winded: one would do `np.all(<comp1> == <comp2>)`

But as noted in the main discussion, I don't think this is the right behaviour for representations - if we implement this at all, they should return arrays of bool, i.e., something like
```
result = True
for compoment in self.components:
    result = np.logical_and(result, getattr(self, component) == getattr(other, component))
```

But I would suggest waiting with making further changes until we have reached some consensus; just returning `False` if the classes are not the same feels wrong to me too. Plus, this doesn't check for derivatives.

Reply:
Yupp, it is very hard to compare representation of two different classes due to rounding error. In that case we have to ignore very slight error. I at first tried to implement by convering both self and other to cartesian.